frontend/src/app/offers/create-offer/create-offer.component.html

```diff
@@ -0,0 +1,295 @@
+      <span class="form-text text-muted" *ngIf="!title.valid &&title.touched">To pole jest obowiązkowe.</span>
+    </div>
+    <div class="col-sm-3">Określ, jak oferta będzie nazywać się w systemie, staraj się być dokładnym.</div>
+  </div>
+  <div class="form-group row">
+    <label for="location" class="col-sm-3 col-form-label">Miejsce odbywania wolontariatu:</label>
+    <div class="col-sm-6">
+      <input 
+        type="text"
+        class="form-control"
+        placeholder="Miejsce odbywania wolontariatu..." 
+        minlength="3"
+        maxlength="150"
+        id="location" name="location"
+        [(ngModel)]="offer.location"
+        #location="ngModel"
+        required>
+      <span class="form-text text-muted" *ngIf="!location.valid &&location.touched">To pole jest obowiązkowe.</span>
+    </div>
+    <div class="col-sm-3">Określ, w jakim miejscu odbywać się będzie wolontariat.</div>
+  </div>
+  <div class="form-group row">
+    <label for="organization" class="col-sm-3 col-form-label">Z ramienia organizacji:</label>
+    <div class="col-sm-6">
+      <select class="form-control"
+        id="organization" name="organization"
+        [(ngModel)]="offer.organization"
+        #organization="ngModel"
+        required>
+        <option  *ngFor="let organization of user['organizations']" [ngValue]="organization" >
+          {{ organization.name }}</option>
+      </select>
+      <span class="form-text text-muted" *ngIf="!organization.valid &&organization.touched">To pole jest obowiązkowe.</span>
+    </div>
+    <div class="col-sm-3">Określ organizację odpowiedzialną za wolontariat.</div>
+  </div>
+  <h3>Realizacja akcji:</h3>
+  <div class="form-group row">
+    <label for="startedAt" class="col-sm-3 col-form-label">Dzień rozpoczęcia:</label>
+    <div class="col-sm-3">
+      <div class="form-check form-check-inline">
+        <div class="input-group">
+          <input 
+            id="startedAt"
+            type="datetime"
+            class="form-control"
+            min="2018-01-01"
+            name="startedAt"
+            [(ngModel)]="offer.startedAt"
+            #startedAt="ngModel"
+            required>
+        </div>
+      </div>
+      <span class="form-text text-muted" *ngIf="startedAt.touched && !startedAt.valid">Podaj datę.</span>
+    </div>
+    <div class="col-sm-3">
+      <div class="form-check form-check-inline">
+          <input
+          type="radio"
+          value=true
+          name="actionOngoing"
+          [(ngModel)]="offer.actionOngoing"
+          #actionOngoing="ngModel">
+          Akcja już trwa
+      </div>
+    </div>
+    <div class="col-sm-3">Określ datę rozpoczęcia akcji.</div>
+  </div>
+  <div class="form-group row ">
+    <label for="finishedAt" class="col-sm-3 col-form-label">Dzień zakończenia:</label>
+    <div class="col-sm-3">
+      <div class="form-check form-check-inline">
+        <div class="input-group">
+          <input 
+            type="datetime"
```

Comment:
I've cloned your pull request to my local git repository and after testing your changes manually I have found that we should probably change the way date is entered by the user. I think we should change all the inputs of type='datetime' and replace them with bootstrap datepicker from ngb module 'ngb-datepicker' . Please see the documentation for datepicker here: [link](https://ng-bootstrap.github.io/#/components/datepicker/examples), I think the best fit would be the example named 'Datepicker in a popup'. 

Also, bear in mind that finishedAt should not be allowed to be set before startedAt, so we need some logic for validating those two inputs together as well.

Reply:
Actually, I have asked @magul privately if we could add this datetime picker package to the project: https://danielykpan.github.io/date-time-picker/, as we need the datetime picker and ng-bootstrap does not provide one - only date picker and time picker separately. If we could use the datetime picker (in range mode preferably) then we could solve the issue with date format. Also, after some reading I have found out that input type='datetime' is deprecated and deleted from HTML specification for more than two years now so we shouldn't use it at all!

Reply:
Let's leave this issue for now, we can fix it in another ticket after you finish this pull request ;)

Reply:
@CorneeSean I am familiar with this issue, I've also talked with magul about that. At first version of this form I've used type=date, but there were issues with backend and we decided it wasn't our priority at the time (but It really drives me mad :P). Thanks for the links, I will check it for sure, but like you said it would be great to finish this one and then to work on that. :) 
Also, there's validation of that startedAt finishedAt thing at the backend side, should I provide it in the front also?

Reply:
TL:DR; We need front-end validation here, but you don't have to do it in this pull request, because we are planning to make important changes to the inputs here anyway  I guess :)

To explain it in a broader context: I guess we would like to have a modern, fluid, user-friendly experience here. Let's give it some perspective first: we are moving from "classical" web portal, where the logic is placed on the server and whole webpages are rendered on the server as well, to a modern web application called sometimes Single Page App (SPA), where most of the user related logic is handled by front-end (here by Angular framework) and most of the data related logic is handled on backend. This way users can feel they are really interacting with the web page instead of just viewing it. However, one of drawbacks of Single Page Apps is that it requires double validation if we want to give it a modern feel.

Validation on front-end side should be instantaneous - user can instantly see if he made a mistake, as after his every action he gets feedback from the app. E.g. if he sets a wrong date, he knows it right away.

To get the same level of feedback from the server, he would need to fill the whole form first, submit it and wait for the server response, which should regenerate the webpage with errors added to the wrongly filled fields. This takes some time and does not feel as good as instantaneous feedback below input he just finished filling.

Why do we still need double validation though? Because backend API should not trust that a front-end request is sent correctly. If there was a mistake in front-end request data and it would not get caught on backend, then our data integrity or security would get compromised and we don't want that to happen. Also, let's imagine for a while that in the near future we would like to open our API to the public and make a public service that anybody can send requests to - then we should not trust those requests at all and we should validate every one of them.

I Hope it got clearer too you now :)

Reply:
@CorneeSean Ok, I get it. :) Thank you. :)